server/src/main/java/org/elasticsearch/cluster/coordination/JoinTaskOutdatedException.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.coordination;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.io.stream.StreamInput;
+
+import java.io.IOException;
+
+/**
+ * Is thrown when a join requests arrives with outdated term
+ */
+public class JoinTaskOutdatedException extends ElasticsearchException {
```

Comment:
I think we can just use `NotMasterException`, no need for a new exception like this.